pkg/generate/cleanup/controller.go

```diff
@@ -338,11 +343,25 @@ func (c *Controller) syncGenerateRequest(key string) error {
 		return err
 	}
 
-	_, err = c.pLister.Get(gr.Spec.Policy)
+	pNamespace, pName, err := cache.SplitMetaNamespaceKey(gr.Spec.Policy)
+	if err != nil {
+		return err
+	}
+	_, err = c.pLister.Get(pName)
 	if err != nil {
 		if !apierrors.IsNotFound(err) {
 			return err
 		}
+
+		_, err = c.npLister.Policies(pNamespace).Get(pName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
+			}
+		} else {
+			return c.processGR(*gr)
+		}
```

Comment:
Should we split this check into its separate logic? Otherwise everytime if the clusterpolicy is not found, this check will be executed which could be unnecessary.

Checking based on `if namespce == ""` makes more sense to me, wdyt?